internal/workflow/processing.go

```diff
@@ -1188,3 +1207,61 @@ func (w *ProcessingWorkflow) sendToFailedBucket(
 
 	return nil
 }
+
+func (w *ProcessingWorkflow) validatePREMIS(
+	ctx temporalsdk_workflow.Context,
+	xmlPath string,
+	paID int,
+) error {
+	id, err := w.createPreservationTask(
+		ctx,
+		datatypes.PreservationTask{
+			Name:                 "Validate PREMIS",
+			Status:               enums.PreservationTaskStatusInProgress,
+			PreservationActionID: paID,
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("create validate PREMIS task: %v", err)
+	}
+
+	pt := datatypes.PreservationTask{
+		ID:     id,
+		Status: enums.PreservationTaskStatusDone,
+		Note:   "PREMIS is valid",
+	}
+
+	// Validate PREMIS.
+	var xmlvalidateResult xmlvalidate.Result
+	activityOpts := withActivityOptsForRequest(ctx)
+	err = temporalsdk_workflow.ExecuteActivity(activityOpts, xmlvalidate.Name, xmlvalidate.Params{
+		XMLPath: xmlPath,
+		XSDPath: w.cfg.ValidatePREMIS.XSDPath,
+	}).Get(activityOpts, &xmlvalidateResult)
+	if err != nil {
+		pt.Status = enums.PreservationTaskStatusError
+		pt.Note = "System error"
+	}
+	if len(xmlvalidateResult.Failures) > 0 {
+		pt.Status = enums.PreservationTaskStatusError
+		var note string
+		for _, failure := range xmlvalidateResult.Failures {
+			note = fmt.Sprintf("%s\n", failure)
+		}
```

Comment:
@mcantelon This is not appending to the note, it will just use the last failure. Also, we should be able to use `pt.Note` directly instead of creating `note`.

Reply:
I've fixed that and simplified it a bit. The display of preservation task notes in Enduro doesn't yet reflect line breaks in the text so the results appear as a jumble, but the info gets there at least.